### PR TITLE
Server query cancellation

### PR DESCRIPTION
**Pull request by vkagamlyk**
Issue #, if available:

Description of changes:

If the user starts a new search without waiting for a result, the request continues to be executed on the server. In this PR I added cancellation of the request on the server for Gremlin and SPARQL

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/graph-explorer/src/connector/openCypher/useOpenCypher.ts
Comment (kmcginnes) [inline packages/graph-explorer/src/connector/openCypher/useOpenCypher.ts:1]: Since we can't set a client generated queryId in OpenCypher, should we undo the changes to this file?

File: packages/graph-explorer/src/connector/useGEFetch.ts [hunk 1]
```diff
@@ -43,6 +43,9 @@ const useGEFetch = () => {
     async (url, options) => {
       const response = await fetch(url, options);
       const data = await response.json();
+      if (typeof options.successCallback === "function") {
+        options.successCallback();
+      }
```
Comment (kmcginnes) [inline packages/graph-explorer/src/connector/useGEFetch.ts:48]: This will miss a couple of scenarios:

- We get a response, but the JSON decoding fails
- The connection is dropped during the `fetch`

Comment (kmcginnes) [reply packages/graph-explorer/src/connector/useGEFetch.ts:48]: One possible solution is to call the callback in all scenarios to ensure the queryId is cleared.

TanStack Query has a similar concept, and they named the callback `onSettled`, which I think is fitting.

https://tanstack.com/query/latest/docs/framework/react/reference/useMutation

File: packages/graph-explorer/src/connector/useGEFetchTypes.ts [hunk 2]
```diff
@@ -7,6 +7,8 @@ import {
 
 export type QueryOptions = RequestInit & {
   disableCache?: boolean;
+  queryId?: string;
+  successCallback?: any;
```
Comment (kmcginnes) [inline packages/graph-explorer/src/connector/useGEFetchTypes.ts:11]: Let's define this type to be the function you expect.

File: packages/graph-explorer/src/connector/useGEFetch.ts (hunk 1 above)
Comment (vkagamlyk) [reply packages/graph-explorer/src/connector/useGEFetch.ts:51]: replaced with try/finally

**Review by kmcginnes — APPROVED** [review]
I have tested the code locally and it looks good.

**Review by michaelnchin — APPROVED** [review]
Thanks @vkagamlyk !
